Makefile

```diff
@@ -12,28 +13,37 @@ build: core $(all_apps)
 .PHONY: clean
 clean: clean-core $(addprefix clean-app-,$(apps))
 
+node_modules: package.json package-lock.json
+	npm install && touch node_modules
```

Comment:
`touch` here makes sure that the mtime of the folder is the most recent as the folder might be created much earlier, this is only for accuracy